api/service/projects_service.go

```diff
@@ -91,15 +103,44 @@ func (service *projectsService) ListProjects(ctx context.Context, name string, u
 	return allProjects, nil
 }
 
-func (service *projectsService) UpdateProject(ctx context.Context, project *models.Project) (*models.Project, error) {
+func (service *projectsService) UpdateProject(ctx context.Context, project *models.Project) (*models.Project, string,
+	error) {
 	if service.authEnabled {
 		err := service.updateAuthorizationPolicy(ctx, project)
 		if err != nil {
-			return nil, fmt.Errorf("error while updating authorization policy for project %s", project.Name)
+			return nil, "", fmt.Errorf("error while updating authorization policy for project %s", project.Name)
+		}
+	}
+
+	updatedProject, err := service.save(project)
+	if err != nil {
+		return nil, "", fmt.Errorf("error saving project %s: %w", project.Name, err)
+	}
+
+	if service.updateProjectConfig.Endpoint != "" &&
+		service.updateProjectConfig.PayloadTemplate != "" &&
+		service.updateProjectConfig.ResponseTemplate != "" {
+
+		payload, err := service.MakeRequestPayload(project, service.updateProjectConfig.PayloadTemplate)
```

Comment:
if these are only used by this service, they should not be a service method, its exposed and not encapsulated as such.